src/parsers/manifestjson.js

```diff
@@ -203,12 +204,29 @@ export default class ManifestJSONParser extends JSONParser {
       baseObject = messages.MANIFEST_FIELD_REQUIRED;
     } else if (error.keyword === 'deprecated') {
       if (
-        this.isStaticTheme &&
-        DEPRECATED_STATIC_THEME_LWT_ALIASES.includes(error.dataPath)
+        Object.prototype.hasOwnProperty.call(
+          DEPRECATED_MANIFEST_PROPERTIES,
+          error.dataPath
+        )
       ) {
-        baseObject = messages.MANIFEST_THEME_LWT_ALIAS;
-        // Overwrite the message with the shorter one included in the linter messages.
+        baseObject = messages[DEPRECATED_MANIFEST_PROPERTIES[error.dataPath]];
+
+        if (baseObject === null) {
+          baseObject = messages.MANIFEST_FIELD_DEPRECATED;
+        }
+
+        let errorDescription = baseObject.description;
+
+        if (errorDescription === null) {
+          const parsedHtml = cheerio.load(error.message);
```

Comment:
sanitize the deprecation message sounds like a good idea to me, but I was wondering if cheerio.load may fail (e.g. by throwing on invalid message) and so if we should then prevent that by:

- run some validation of the deprecated messages while we are importing the schema
- or pre-sanitize all the deprecated messages as part of the schema importing

given that we already have a schema importing (and transformation) step, pre-sanitize the deprecated messages as part of that step may be good (even if cheerio.load would not throw on an invalid HTML fragment included in these messages).

Reply:
Deferring this to a follow up sounds ok to me (as we discussed over IRC).
have you already filed a follow up issue to pre-sanitize (or validate) the deprecated messages as part of the schema importing? it would be good to link it here.

it may still be good to cover this part with a couple of ` errorLookup` unit tests (in the `test.manifestjson.js` test file).

I also took a quick look to the cheerio npm dependency and the API schemas in the mozilla-central repo to see if there is any deprecation message that could be able to trigger an exception here and it doesn't seem there is any:

it seems that `cheerio.load` may only throw is `error.message` is not a string (and looking at the searchfox results this one is the only one to be a boolean: https://searchfox.org/mozilla-central/rev/9bb55ae4d808fc48afcf93f99da6a685265b86c6/browser/components/extensions/schemas/windows.json#114, it is not from an API property and so it shouldn't be able to trigger an exception here)

Reply:
As a side note:
there are some other annoying "message formats" that we inherited from chrome, e.g. `Please use $(ref:tabs.onActivated)`, but it seems that none of the deprecation messages related to manifest properties is using it (https://searchfox.org/mozilla-central/search?q=%22deprecated%22%3A&case=false&regexp=false&path=schemas%2F) and so it shouldn't be actually an issue yet.